pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -460,89 +474,76 @@ func (r Client) removePrecopies(precopies []planapi.Precopy, vmService *ovirtsdk
-				Factor:   0,
-				Jitter:   0,
-				Steps:    3,
-			}
-
-			err = wait.ExponentialBackoff(backoff, func() (bool, error) {
-				snapshotDisksResponse, err := snapService.DisksService().List().Send()
-				if err != nil {
-					r.Log.Error(err, "snapshotDiksService Request failed")
-					err = liberr.Wrap(err, "Disks not found")
-					return false, err
-				}
-				snapshotDisks, found := snapshotDisksResponse.Disks()
-				if !found {
-					r.Log.Error(err, "disks not found")
-					err = liberr.Wrap(err, "Disks not found")
-					return false, err
-				}
-				for _, disk := range snapshotDisks.Slice() {
-					if transferred, err := r.isDiskBeingTransferred(disk.MustId()); transferred {
-						r.Log.Info("Disk is being transferred, retrying...")
-						return false, nil
-					} else if err != nil {
-						err = liberr.Wrap(err)
-						r.Log.Error(err, "Error checking if disk is being transferred")
-						return false, nil
-					} else {
-						// Try to remove the snapshot again
-						_, err = snapService.Remove().Query("correlation_id", correlationID).Send()
-						if err != nil {
-							err = liberr.Wrap(err)
-							r.Log.Error(err, "Error removing snapshot")
-							return true, err
-						}
-					}
-				}
-				return false, nil
-			})
-			if err != nil {
-				err = liberr.Wrap(err)
-				r.Log.Error(err, "Error waiting for snapshot removal")
-				return
-			}
-
-		}
-
 		for {
-			jobs, err := r.getJobs(correlationID)
-			if err != nil {
-				err = liberr.Wrap(err)
-				r.Log.Error(err, "Error waiting for snapshot removal")
-				break
-			}
-			if allJobsFinished(jobs) {
-				break
-			}
-
 			select {
 			case <-time.After(time.Duration(settings.Settings.Migration.SnapshotRemovalTimeout) * time.Minute):
 				r.Log.Info("Timeout waiting for snapshot removal")
 				return
 			default:
 				time.Sleep(time.Duration(settings.Settings.Migration.SnapshotStatusCheckRate) * time.Second)
 			}
+			// Try to remove the snapshot
+			_, err := snapService.Remove().Query("correlation_id", correlationID).Send()
+			if err != nil {
+				if strings.Contains(err.Error(), "Cannot remove Snapshot") {
+					err = web.ConflictError{
+						Provider: r.Source.Provider,
+						Err:      err,
+					}
+					r.Log.Error(err, "ConflictError failed to remove snapshot", "snapshotID", snapshotID)
+				} else {
+					err = liberr.Wrap(err)
+					r.Log.Error(err, "Remove request for snapshot failed", "snapshotID", snapshotID)
+
+					snapshotDisksResponse, err := snapService.DisksService().List().Send()
+					if err != nil {
+						r.Log.Error(err, "Request for snapshotDisksResponse failed")
+						break
+					}
+					snapshotDisks, found := snapshotDisksResponse.Disks()
+					if !found {
+						r.Log.Error(err, "Disks not found")
+						continue
+					}
+					for _, disk := range snapshotDisks.Slice() {
+						if transferred, err := r.isDiskBeingTransferred(disk.MustId()); transferred {
+							r.Log.Info("Disk is being transferred, retrying...", "disk", disk)
```

Comment:
I think we won't get here anymore (need to check) but if we do then we need to return ConflictError to keep trying

Reply:
yes here is a bit different control logic compared to the creation.
Here is everything controlled by the `continue` and `break` so we don't need the `ConflictError`. I'll add the `continue` to jump to the next retry.

Reply:
@mnecas right but I'd expect remove-snapshot to return a validation error when the disk is being transferred and then `strings.Contains(err.Error(), "Cannot remove Snapshot")` should be true

Reply:
yeah that's true, I never got to the "disk is being transferred..." part always failed in the ConflictError. I'll remove that part

Reply:
cool, yeah, I also checked it in ovirt-engine - transfer-image locks the vm with a shared lock and remove-snapshot tries to acquire exclusive lock on the vm - so that makes sense